### PR TITLE
Convert planners module to C++

### PR DESCRIPTION
**Pull request by ecourtois**
Create a virtual Planner class defining common methods and pure virtual
methods required in all planner implementations.
Create a class for Astar planner that inherit the virtual Planner class.

### CODE REVIEW

File: planners/astar/AstarPlanner.cpp [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+        }
+        if (nb_pose_reachable < 0) {
+            DEBUG("planner: No position reachable!\n");
+            goto trajectory_get_route_update_error;
+        }
+
+        avoidance_index = 1;
+    }
+
+    /* At this point the avoidance graph is valid and the next targeted
```
Comment (gdoffe) [inline planners/astar/AstarPlanner.cpp:135]: change /* */ to //

File: planners/astar/AstarPlanner.cpp [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+     * point can be retrieved. */
+    pose_to_reach = avoidance_get_pose(avoidance_index);
+
+    /* Check if the point to reach is the current path position to reach */
```
Comment (gdoffe) [inline planners/astar/AstarPlanner.cpp:139]: change /* */ to //

File: planners/astar/AstarPlanner.cpp [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+    for (;;) {
+        riot::time_point loop_start_time = riot::now();
+
+        if (!started_) {
+            goto yield_point;
+        }
+
+        current_path_pos = path_.current_pose();
+
+        /* Update speed order to max speed defined value in the new point to reach */
```
Comment (gdoffe) [inline planners/astar/AstarPlanner.cpp:185]: change /* */ to //

File: planners/astar/AstarPlanner.cpp [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+        speed_order.set_distance(path_.current_max_speed());
+        speed_order.set_angle(speed_order.distance() / 2);
+
+        /* reverse gear selection is granted per point to reach, in path */
```
Comment (gdoffe) [inline planners/astar/AstarPlanner.cpp:189]: change /* */ to //

**Review by gdoffe — APPROVED** [review]
> Create a class for Astar planner that inherit**s** the virtual Planner class.

Very good PR, I love it ! I approve and let you fix cosmetic stuff, no need to re-review

File: planners/astar/AstarPlanner.cpp (hunk 4 above)
Comment (ecourtois) [reply planners/astar/AstarPlanner.cpp]: Done

File: planners/astar/AstarPlanner.cpp (hunk 3 above)
Comment (ecourtois) [reply planners/astar/AstarPlanner.cpp]: Done

File: planners/astar/AstarPlanner.cpp (hunk 2 above)
Comment (ecourtois) [reply planners/astar/AstarPlanner.cpp]: Done

File: planners/astar/AstarPlanner.cpp (hunk 1 above)
Comment (ecourtois) [reply planners/astar/AstarPlanner.cpp]: Done

**Pull request merged**